app/notification_email.go

```diff
@@ -284,6 +288,23 @@ func getFormattedPostTime(user *model.User, post *model.Post, useMilitaryTime bo
 	}
 }
 
+func (a *App) generateHyperlinkForChannels(postMessage, teamName, teamURL string) string {
+	r := regexp.MustCompile(`\~[-a-zA-Z0-9]+\b`)
```

Comment:
This is related to @jespino's comment about using `GetMessageForNotification`, but this is going to cause conflicts with a few different Markdown things. The two that I can think of are:
1. Channel links inside of code blocks will be rendered as actual links
2. Strikethrough text such as ~~this~~ (`~~this~~`) may be detected as potential channel links

Since we don't support other Markdown in these emails yet, this might be fine for now, but I think it's something to be aware of.